lib/components/user/notification-prefs-pane.js

```diff
@@ -85,13 +223,60 @@ class NotificationPrefsPane extends Component {
             </FormGroup>
           )}
           {notificationChannel === 'sms' && (
-            <FormGroup>
-              <ControlLabel>Enter your phone number for SMS notifications:</ControlLabel>
-              {/* TODO: Add field validation. */}
-              <FormControl onChange={this._handlePhoneNumberChange} type='tel' value={phoneNumber} />
-            </FormGroup>
+            <div>
+              <FormGroup validationState={phoneFieldValidationState}>
+                <ControlLabel>Enter your phone number for SMS notifications:</ControlLabel>
+                {/* TODO: Add field validation. */}
+
+                <FormControl
+                  onChange={this._handlePhoneNumberChange}
+                  // Fake phone number, see https://en.wikipedia.org/wiki/555_(telephone_number)
+                  placeholder='+15555550123'
+                  type='tel'
+                  value={phoneNumber}
+                />
+                {/* Show glyphs underneath the input control with <HelpBlock>
+                   (there are some alignment issues with <FormControl.Feedback /> in mobile view). */}
+                <HelpBlock>
+                  {phoneStatusGlyph && <Glyphicon glyph={phoneStatusGlyph} />} {phoneStatusText}
+                </HelpBlock>
+              </FormGroup>
+
+              {shouldVerifyPhone && (
+                <ButtonStrip>
+                  <Button bsStyle='primary' onClick={this._handleRequestPhoneVerificationCode}>Verify my phone</Button>
+                  {initialPhoneNumber.length !== 0 && <Button onClick={this._handleRevertPhoneNumber}>Revert number</Button>}
+                </ButtonStrip>
+              )}
+            </div>
           )}
         </Details>
+
+        {/* The dialog prompt for validation code. */}
+        <Modal onHide={this._handlePhoneValidationCancel} show={isVerificationPending}>
+          <Modal.Header>
+            <Modal.Title>Enter verification code</Modal.Title>
+          </Modal.Header>
+
+          <Modal.Body>
+            <p>
+              Please check the SMS messaging app on your mobile phone
+              for a text message with a verification code, and enter the code below.
+            </p>
+            <FormGroup>
+              <ControlLabel>Verification code:</ControlLabel>
+              <FormControl onChange={this._handlePhoneValidationCodeChange} value={phoneValidationCode} />
+            </FormGroup>
+            <p>
+              <Button bsStyle='link' onClick={this._handleRequestPhoneVerificationCode}>Request a new code</Button>
```

Comment:
Left padding makes this button offset in a weird way.

Reply:
Fixed